workflow-templates/phpunit-mariadb.yml

```diff
@@ -0,0 +1,180 @@
+# This workflow is provided via the organization template repository
+#
+# https://github.com/nextcloud/.github
+# https://docs.github.com/en/actions/learn-github-actions/sharing-workflows-with-your-organization
+
+name: PHPUnit mariadb
+
+on:
+  pull_request:
+  push:
+    branches:
+      - main
+      - master
+      - stable*
+
+permissions:
+  contents: read
+
+concurrency:
+  group: phpunit-mariadb-${{ github.head_ref || github.run_id }}
+  cancel-in-progress: true
+
+jobs:
+  changes:
+    runs-on: ubuntu-latest
+
+    outputs:
+      src: ${{ steps.changes.outputs.src}}
+
+    steps:
+      - uses: dorny/paths-filter@4512585405083f25c027a35db413c2b3b9006d50 # v2.11.1
+        id: changes
+        continue-on-error: true
+        with:
+          filters: |
+            src:
+              - '.github/workflows/**'
+              - 'appinfo/**'
+              - 'lib/**'
+              - 'templates/**'
+              - 'tests/**'
+              - 'vendor/**'
+              - 'vendor-bin/**'
+              - '.php-cs-fixer.dist.php'
+              - 'composer.json'
+              - 'composer.lock'
+
+  phpunit-mariadb:
+    runs-on: ubuntu-latest
+
+    needs: changes
+    if: needs.changes.outputs.src != 'false'
+
+    strategy:
+      matrix:
+        php-versions: ['8.0', '8.1', '8.2', '8.3']
```

Comment:
This is quite a big matrix. Maybe only test with one PHP version like with sqlite, oci and postgres?

Reply:
For now I changed this, but my question is: What is the recommended database to test all PHP versions?
Currently it seems to be MySQL, but maybe we should set the default to sqlite to save some CI time (as it is faster, but  you probably want to at least test your app with every PHP version once and there is only one SQLite version)

Reply:
It's mysql/mariadb and should be either of those or postgres. Sqlite is not typing at all, so more likely to allow buggy code/queries in general.
If we optimize that to the one db no one should use in production we are optimizing the wrong thing or based on the wrong matrix.

I would keep it on the mysql workflow for now, so people updating that one don't accidentally drop all php matrix testing 😅